sql/src/main/java/io/crate/execution/engine/window/WindowFrameState.java

```diff
@@ -64,14 +68,23 @@ public int size() {
      * Return the row at the given index in the frame or null if the index is out of bounds.
      */
     @Nullable
-    public Object[] getRowAtIndexOrNull(int index) {
+    public Object[] getRowInFrameAtIndexOrNull(int index) {
         if (index < lowerBound || index >= upperBoundExclusive) {
             return null;
         }
         int globalIdx = partitionStart + index;
         return rows.get(globalIdx);
     }
 
+    @Nullable
+    Object[] getRowInPartitionAtIndexOrNull(int index) {
```

Comment:
This now exposes logic that spans out of a frame. I'm not sure if extending the model to have a WindowPartitionState that will contain a "current frame state" in some way (via methods probably, not another class) is worth it. Thoughts?

Reply:
I think it is fine this way.